web/src/app/guards/auth.guard.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import { tap, map, take } from 'rxjs/operators';
+import { AuthService } from '../services/auth.service';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuardService implements CanActivate {
+  constructor(
+    private  auth: AuthService,
+    private router: Router,
+  ) { }
+
+  canActivate(): Observable<boolean> {
+
+    return this.auth.user.pipe(
+      take(1),
+      map(authUser => {
+        console.log('user in map', authUser);
+        return !!authUser;
+      }),
+      tap(loggedIn => {
+        console.log('loggedIn', loggedIn);
```

Comment:
Are these `console.log` debug code? Will they be removed later on?

Reply:
yes @eddiejaoude , They are for debugging twitter format response. Will remove it.

Reply:
Ok 👍 . Do you still need them now? If not can we remove. But if you still need, we can remove later but please add a `@TODO: remove` so we dont forget. Thank you 👍